usr/lib/cca_stdll/cca_specific.c

```diff
@@ -3321,6 +3321,65 @@ CK_RV token_specific_set_attrs_for_new_object(STDLL_TokData_t *tokdata,
 
     return ret;
 }
+
+/*
+ * ECC private key section (X'20'), Key-usage and translation control flag.
+ */
+#define CCA_XPRTCPAC                          0x01
+#define CCA_ECC_TOKEN_KEYUSAGE_OFFSET         16
+
+static CK_BBOOL ccatok_token_is_cpacf_exportable(const CK_BYTE *token)
+{
+    CK_BYTE keyusage = token[CCA_ECC_TOKEN_KEYUSAGE_OFFSET];
+
+    if (keyusage & CCA_XPRTCPAC)
+        return CK_TRUE;
+
+    return CK_FALSE;
+}
+
+static CK_RV ccatok_pkey_check_attrs(STDLL_TokData_t *tokdata,
+                                     TEMPLATE * templ,
+                                     enum cca_token_type token_type)
+{
+    CK_ATTRIBUTE *pkey_attr = NULL, *skey_attr = NULL;
+    CK_RV ret;
+
+    UNUSED(tokdata);
+
+    /* OK, if there's no secure key token at all */
+    ret = template_attribute_find(templ, CKA_IBM_OPAQUE, &skey_attr);
+    if (ret != TRUE)
+        return CKR_OK;
+
+    /* Also ok, if there's no CKA_IBM_PROTKEY_EXTRACTABLE attribute */
+    ret = template_attribute_find(templ, CKA_IBM_PROTKEY_EXTRACTABLE, &pkey_attr);
+    if (ret != TRUE || *(CK_BBOOL *)pkey_attr->pValue == FALSE)
+        return CKR_OK;
+
+    /*
+     * At this point, the key has CKA_IBM_PROKEY_EXTRACTABLE = true and it has
+     * a secure key token, so let's check if the secure key token is
+     * CPACF-exportable.
+     */
+    switch (token_type) {
+    case sec_aes_data_key:
+        /* Nothing to do: AES data keys are always CPACF-exportable */
+        break;
+    case sec_ecc_priv_key:
+        if (pkey_op_ec_curve_supported_by_cpacf(templ) &&
```

Comment:
Do we really want to check the curve here? What if at a alter time CPACF supports more curves? 
I guess we can accept to have CKA_IBM_PROTKEY_EXTRACTABLE=TRUE in any case, except the key blob does not have XPRTCPAC. Thats the only real reason why to reject CKA_IBM_PROTKEY_EXTRACTABLE=TRUE. 

We also allow DES or RSA keys to have CKA_IBM_PROTKEY_EXTRACTABLE=TRUE, although we don't support PKEY for those key types.

Reply:
As discussed offline: The CCA lib allows the XPRTCPAC rule_array keyword only for the 5 curves supported by CPACF: p256, p384, p521, ed25519, ed448. Adjusted the check, but still checking the curves here. 
Done.